add(doc): add schema documentation

ChrisLatRNCan opened the pull request:
Add first draft of the schema documentation
Still missing parts of the map section.
Other updates will be done through #321 

Closes #8, #26

## Description
<!-- Link to an issue or include a description -->

## Testing
<!-- Have you added unit tests for this code?  If not explain why. -->

## Documentation
Wiki

## Checklist
<!-- Quick checklist for items that are easy to miss -->

- [x] Commit messages follow [the guidelines](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/master/CONTRIBUTING.md#-git-commit-guidelines)
- [x] Release notes have been updated
- [x] PR targets the correct release version
- [x] Help files and documentation have been updated

Remember, it is a *muffin offence* to open a PR with any of the above checklist items incomplete.

Please keep the original issue up to date with the final solution, expected behaviour, and any additional notes for testers

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/fgpv-vpgf/fgpa-apgf/325)
<!-- Reviewable:end -->

jolevesq commented:
Very good job! Only few comments...

---

Reviewed 28 of 28 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---

*[wiki/schema.en-CA.md, line 43 at r1](https://reviewable.io/reviews/fgpv-vpgf/fgpa-apgf/325#-LBQhQrc6KlipOhXQkaY:-LBQhQrc6KlipOhXQkaZ:b-xrwd75) ([raw file](https://github.com/fgpv-vpgf/fgpa-apgf/blob/0eac9d3b4be8e77360009236a29953e46d2a3bea/wiki/schema.en-CA.md#L43)):*
> ```Markdown
> &emsp;&emsp;&emsp;],<br/>
> &emsp;&emsp;&emsp;[**"components"**](#mapcomponents): {<br/>
> &emsp;&emsp;&emsp;&emsp;[**"geoSearch"**](#mapcomponentsgeosearch): {<br/>
> ```

All this section mapcomponentgeosearch is not use by the viewer

---

*[wiki/schema.en-CA.md, line 167 at r1](https://reviewable.io/reviews/fgpv-vpgf/fgpa-apgf/325#-LBQhgo34oJPzAhCTINg:-LBQhgo34oJPzAhCTINh:b-8nnkkx) ([raw file](https://github.com/fgpv-vpgf/fgpa-apgf/blob/0eac9d3b4be8e77360009236a29953e46d2a3bea/wiki/schema.en-CA.md#L167)):*
> ```Markdown
> [:arrow_up: back to Map](#map-json-tree)
> 
> >### [map:components:_**`geoSearch`**_](#map--components--geosearch)
> ```

Not used

---

*[wiki/schema.en-CA.md, line 176 at r1](https://reviewable.io/reviews/fgpv-vpgf/fgpa-apgf/325#-LBQhjMZCq9J9MCnJ5Qz:-LBQhjMZCq9J9MCnJ5R-:b-8nnkkx) ([raw file](https://github.com/fgpv-vpgf/fgpa-apgf/blob/0eac9d3b4be8e77360009236a29953e46d2a3bea/wiki/schema.en-CA.md#L176)):*
> ```Markdown
> [:arrow_up: back to Map](#map-json-tree)
> 
> >### [map:components:geoSearch:_**`enabled`**_](#map--components--geosearch--enabled)
> ```

Not used

---

*[wiki/schema.en-CA.md, line 185 at r1](https://reviewable.io/reviews/fgpv-vpgf/fgpa-apgf/325#-LBQhkzk4JNo8tYiCa-2:-LBQhkzk4JNo8tYiCa-3:b-8nnkkx) ([raw file](https://github.com/fgpv-vpgf/fgpa-apgf/blob/0eac9d3b4be8e77360009236a29953e46d2a3bea/wiki/schema.en-CA.md#L185)):*
> ```Markdown
> [:arrow_up: back to Map](#map-json-tree)
> 
> >### [map:components:geoSearch:_**`showGraphic`**_](#map--components--geosearch--showgraphic)
> ```

Not used

---

*[wiki/schema.en-CA.md, line 194 at r1](https://reviewable.io/reviews/fgpv-vpgf/fgpa-apgf/325#-LBQhojM0YWUheKVT6bW:-LBQhojM0YWUheKVT6bX:b-8nnkkx) ([raw file](https://github.com/fgpv-vpgf/fgpa-apgf/blob/0eac9d3b4be8e77360009236a29953e46d2a3bea/wiki/schema.en-CA.md#L194)):*
> ```Markdown
> [:arrow_up: back to Map](#map-json-tree)
> 
> >### [map:components:geoSearch:_**`showInfo`**_](#map--components--geosearch--showinfo)
> ```

Not used

---

*[wiki/schema.en-CA.md, line 272 at r1](https://reviewable.io/reviews/fgpv-vpgf/fgpa-apgf/325#-LBQiDFlC6mY6c9rgdqR:-LBQiDFlC6mY6c9rgdqS:bpcg50d) ([raw file](https://github.com/fgpv-vpgf/fgpa-apgf/blob/0eac9d3b4be8e77360009236a29953e46d2a3bea/wiki/schema.en-CA.md#L272)):*
> ```Markdown
> | ------- | ------- | ------- | ------- | ------- | ------- | ------- | ------- |
> | [**overviewMap : enabled**](#map--components--overviewmap)| *boolean* |  | - | `true` | Map/Components | - | - |
> | [**overviewMap : expandFactor**](#map--components--overviewmap) | *number* |  | `2` | `5` | `true` | Map/Components | - | - |
> ```

Question... for overview map and scalebar you put properties inside a table but not for North Arrow and Mouseinfo. Is there a reason? I like when it is inside a table. It reduce the amount of link on the page and it is easier to read.

Do you have to repeat overviewMap : ... before the properties?

---

*[wiki/schema.en-CA.md, line 624 at r1](https://reviewable.io/reviews/fgpv-vpgf/fgpa-apgf/325#-LBQk3S-B0TyIuNJIt88:-LBQk3S-B0TyIuNJIt89:b-6t6cue) ([raw file](https://github.com/fgpv-vpgf/fgpa-apgf/blob/0eac9d3b4be8e77360009236a29953e46d2a3bea/wiki/schema.en-CA.md#L624)):*
> ```Markdown
> [:arrow_up: back to UI](#ui-json-tree)
> 
> >### [ui:_**`appBar`**_](#ui--appbar)
> ```

When it is the last level should it be inside the table

AppBar
___________
geosearch | boolean | ...
___________
SideMenu | boolean | ...

To reduce the amount of links

---

*[wiki/schema.en-CA.md, line 633 at r1](https://reviewable.io/reviews/fgpv-vpgf/fgpa-apgf/325#-LBQl02z9NANDgJbvjZ5:-LBQl02z9NANDgJbvjZ6:bjuamr9) ([raw file](https://github.com/fgpv-vpgf/fgpa-apgf/blob/0eac9d3b4be8e77360009236a29953e46d2a3bea/wiki/schema.en-CA.md#L633)):*
> ```Markdown
> [:arrow_up: back to UI](#ui-json-tree)
> 
> >### [ui:appBar:_**`sideMenu`**_](#ui--appbar)
> ```

When we click on a link, can we scroll a little bit so the item is not at the very top but in the middle of the screen?

---

*[wiki/schema.en-CA.md, line 824 at r1](https://reviewable.io/reviews/fgpv-vpgf/fgpa-apgf/325#-LBQlqCHEE54xVHn1dP3:-LBQlqCHEE54xVHn1dP4:b-s6k39k) ([raw file](https://github.com/fgpv-vpgf/fgpa-apgf/blob/0eac9d3b4be8e77360009236a29953e46d2a3bea/wiki/schema.en-CA.md#L824)):*
> ```Markdown
> 
> >### [ui:_**`tableIsOpen`**_](#ui--tableisopen)
> >>A set of service endpoints used by the viewer.
> ```

From all the section title can we have a link to go to the main top of the page?

---

*[wiki/schema.en-CA.md, line 1414 at r1](https://reviewable.io/reviews/fgpv-vpgf/fgpa-apgf/325#-LBQmJo7C-idYUXKiK1D:-LBQmJo7C-idYUXKiK1E:bcwyqa0) ([raw file](https://github.com/fgpv-vpgf/fgpa-apgf/blob/0eac9d3b4be8e77360009236a29953e46d2a3bea/wiki/schema.en-CA.md#L1414)):*
> ```Markdown
> >## **services : exportMapUrl**
> 
> You can find more information about **exporting map to image services** [here](https://developers.arcgis.com/rest/services-reference/export-web-map-task.htm).
> ```

When you are redirected to a new window, can you open a new tab instead of leaving the page?

---

*[wiki/schema.en-CA.md, line 1495 at r1](https://reviewable.io/reviews/fgpv-vpgf/fgpa-apgf/325#-LBQlURo2i-Ypic9hOVV:-LBQlURo2i-Ypic9hOVW:b-1e35nw) ([raw file](https://github.com/fgpv-vpgf/fgpa-apgf/blob/0eac9d3b4be8e77360009236a29953e46d2a3bea/wiki/schema.en-CA.md#L1495)):*
> ```Markdown
> Default search services are provided by Natural Resources Canada and cover Canadian territories. You can find more information about thoses services here:
> 
> [GeoLocation API](https://www.nrcan.gc.ca/earth-sciences/geography/topographic-information/free-data-geogratis/geogratis-web-services/17216#g3): **"geoLocation"** and **"geoSuggets"**
> ```

Is it a typo... suggest?

---


*Comments from [Reviewable](https://reviewable.io/reviews/fgpv-vpgf/fgpa-apgf/325#-:-LBQnBonEmr6YOg-WOGt:b1slhq4)*
<!-- Sent from Reviewable.io -->

ChrisLatRNCan commented:
Review status: 26 of 28 files reviewed at latest revision, 11 unresolved discussions.

---

*[wiki/schema.en-CA.md, line 43 at r1](https://reviewable.io/reviews/fgpv-vpgf/fgpa-apgf/325#-LBQhQrc6KlipOhXQkaY:-LBQrEoW5M2YrqZj8mLX:b-896fix) ([raw file](https://github.com/fgpv-vpgf/fgpa-apgf/blob/0eac9d3b4be8e77360009236a29953e46d2a3bea/wiki/schema.en-CA.md#L43)):*
<details><summary><i>Previously, jolevesq (Johann Levesque) wrote…</i></summary><blockquote>

All this section mapcomponentgeosearch is not use by the viewer
</blockquote></details>

Done.

---

*[wiki/schema.en-CA.md, line 167 at r1](https://reviewable.io/reviews/fgpv-vpgf/fgpa-apgf/325#-LBQhgo34oJPzAhCTINg:-LBQrFrH4PvcXmsmku5I:b-896fix) ([raw file](https://github.com/fgpv-vpgf/fgpa-apgf/blob/0eac9d3b4be8e77360009236a29953e46d2a3bea/wiki/schema.en-CA.md#L167)):*
<details><summary><i>Previously, jolevesq (Johann Levesque) wrote…</i></summary><blockquote>

Not used
</blockquote></details>

Done.

---

*[wiki/schema.en-CA.md, line 176 at r1](https://reviewable.io/reviews/fgpv-vpgf/fgpa-apgf/325#-LBQhjMZCq9J9MCnJ5Qz:-LBQrGZd0YvrBj04BoLB:b-896fix) ([raw file](https://github.com/fgpv-vpgf/fgpa-apgf/blob/0eac9d3b4be8e77360009236a29953e46d2a3bea/wiki/schema.en-CA.md#L176)):*
<details><summary><i>Previously, jolevesq (Johann Levesque) wrote…</i></summary><blockquote>

Not used
</blockquote></details>

Done.

---

*[wiki/schema.en-CA.md, line 185 at r1](https://reviewable.io/reviews/fgpv-vpgf/fgpa-apgf/325#-LBQhkzk4JNo8tYiCa-2:-LBQrI-WCFeqwHu_xir2:b-896fix) ([raw file](https://github.com/fgpv-vpgf/fgpa-apgf/blob/0eac9d3b4be8e77360009236a29953e46d2a3bea/wiki/schema.en-CA.md#L185)):*
<details><summary><i>Previously, jolevesq (Johann Levesque) wrote…</i></summary><blockquote>

Not used
</blockquote></details>

Done.

---

*[wiki/schema.en-CA.md, line 194 at r1](https://reviewable.io/reviews/fgpv-vpgf/fgpa-apgf/325#-LBQhojM0YWUheKVT6bW:-LBQrIkG7r1HFxdONyPD:b-896fix) ([raw file](https://github.com/fgpv-vpgf/fgpa-apgf/blob/0eac9d3b4be8e77360009236a29953e46d2a3bea/wiki/schema.en-CA.md#L194)):*
<details><summary><i>Previously, jolevesq (Johann Levesque) wrote…</i></summary><blockquote>

Not used
</blockquote></details>

Done.

---

*[wiki/schema.en-CA.md, line 272 at r1](https://reviewable.io/reviews/fgpv-vpgf/fgpa-apgf/325#-LBQiDFlC6mY6c9rgdqR:-LBQrT2mDnJhs5C1ptFq:bnqy91v) ([raw file](https://github.com/fgpv-vpgf/fgpa-apgf/blob/0eac9d3b4be8e77360009236a29953e46d2a3bea/wiki/schema.en-CA.md#L272)):*
<details><summary><i>Previously, jolevesq (Johann Levesque) wrote…</i></summary><blockquote>

Question... for overview map and scalebar you put properties inside a table but not for North Arrow and Mouseinfo. Is there a reason? I like when it is inside a table. It reduce the amount of link on the page and it is easier to read.

Do you have to repeat overviewMap : ... before the properties?
</blockquote></details>

Maybe laziness.  If there is no description I may put it this way. But as soon as there's a relatively long description the table begins to be huge vertically and markdown is not flexible enough to correct that (unless I use html). I'm taking note of it in [#321](https://github.com/fgpv-vpgf/fgpa-apgf/issues/321)

---

*[wiki/schema.en-CA.md, line 624 at r1](https://reviewable.io/reviews/fgpv-vpgf/fgpa-apgf/325#-LBQk3S-B0TyIuNJIt88:-LBQuRbaB5DosT1KXWV4:besl43q) ([raw file](https://github.com/fgpv-vpgf/fgpa-apgf/blob/0eac9d3b4be8e77360009236a29953e46d2a3bea/wiki/schema.en-CA.md#L624)):*
<details><summary><i>Previously, jolevesq (Johann Levesque) wrote…</i></summary><blockquote>

When it is the last level should it be inside the table

AppBar
___________
geosearch | boolean | ...
___________
SideMenu | boolean | ...

To reduce the amount of links
</blockquote></details>

Good suggestion.  Added to [#321](https://github.com/fgpv-vpgf/fgpa-apgf/issues/321).

---

*[wiki/schema.en-CA.md, line 633 at r1](https://reviewable.io/reviews/fgpv-vpgf/fgpa-apgf/325#-LBQl02z9NANDgJbvjZ5:-LBQtnF0Cgh0tc7LrtZU:bslyh6m) ([raw file](https://github.com/fgpv-vpgf/fgpa-apgf/blob/0eac9d3b4be8e77360009236a29953e46d2a3bea/wiki/schema.en-CA.md#L633)):*
<details><summary><i>Previously, jolevesq (Johann Levesque) wrote…</i></summary><blockquote>

When we click on a link, can we scroll a little bit so the item is not at the very top but in the middle of the screen?
</blockquote></details>

Humm... Would probably need html/css work. I'll document it in [#321](https://github.com/fgpv-vpgf/fgpa-apgf/issues/321).

---

*[wiki/schema.en-CA.md, line 824 at r1](https://reviewable.io/reviews/fgpv-vpgf/fgpa-apgf/325#-LBQlqCHEE54xVHn1dP3:-LBR-X_N0SHF9DTPMY80:b2shnq3) ([raw file](https://github.com/fgpv-vpgf/fgpa-apgf/blob/0eac9d3b4be8e77360009236a29953e46d2a3bea/wiki/schema.en-CA.md#L824)):*
<details><summary><i>Previously, jolevesq (Johann Levesque) wrote…</i></summary><blockquote>

From all the section title can we have a link to go to the main top of the page?
</blockquote></details>

I'll use the side menu to divide even more the DOC. Added to [#321](https://github.com/fgpv-vpgf/fgpa-apgf/issues/321).

---

*[wiki/schema.en-CA.md, line 1414 at r1](https://reviewable.io/reviews/fgpv-vpgf/fgpa-apgf/325#-LBQmJo7C-idYUXKiK1D:-LBQx9tF-R4TNuDGd1dG:b-ghnkh5) ([raw file](https://github.com/fgpv-vpgf/fgpa-apgf/blob/0eac9d3b4be8e77360009236a29953e46d2a3bea/wiki/schema.en-CA.md#L1414)):*
<details><summary><i>Previously, jolevesq (Johann Levesque) wrote…</i></summary><blockquote>

When you are redirected to a new window, can you open a new tab instead of leaving the page?
</blockquote></details>

Not with markdown. Should work with html but GitHub replace my target="_blank" with rel="nofollow"

---

*[wiki/schema.en-CA.md, line 1495 at r1](https://reviewable.io/reviews/fgpv-vpgf/fgpa-apgf/325#-LBQlURo2i-Ypic9hOVV:-LBQvr6S70fvmT0K_MRB:b-wottgf) ([raw file](https://github.com/fgpv-vpgf/fgpa-apgf/blob/0eac9d3b4be8e77360009236a29953e46d2a3bea/wiki/schema.en-CA.md#L1495)):*
<details><summary><i>Previously, jolevesq (Johann Levesque) wrote…</i></summary><blockquote>

Is it a typo... suggest?
</blockquote></details>

Absolutely, good catch Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/fgpv-vpgf/fgpa-apgf/325)*
<!-- Sent from Reviewable.io -->

jolevesq commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io/reviews/fgpv-vpgf/fgpa-apgf/325#-:-LBRFHzpDH1Y_iU5dp1Y:bnfp4nl)*
<!-- Sent from Reviewable.io -->
